app/vmselect/main.go

```diff
@@ -182,6 +180,61 @@ var (
 	})
 )
 
+func updateSlowQueryPathCounter(p *httpserver.Path, path string) {
```

Comment:
I don't think we need a duplicated function to identify the query path, we can increase counters in the [original handler](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/66fbea70a59c8a04793f81aa2accb6099addd31f/app/vmselect/main.go#L185).
The code can be like this
```
	var slowQueries *metrics.Counter
	if *logSlowQueryDuration > 0 {
		actualStartTime := time.Now()
		defer func() {
			d := time.Since(actualStartTime)
			if d >= *logSlowQueryDuration {
...
				slowQueries.Inc()
			}
		}()
	}
...
	if path == "/admin/tenants" {
		tenantsRequests.Inc()
		slowQueries = pathSlowTenantsRequests
...

// selectHandler()
	switch p.Suffix {
	case "prometheus/api/v1/query":
		queryRequests.Inc()
		slowQueries = pathSlowQuery
		httpserver.EnableCORS(w, r)
		if err := prometheus.QueryHandler(qt, startTime, at, w, r); err != nil {
			queryErrors.Inc()
			sendPrometheusError(w, r, err)
			return true
		}
		return true
	case "prometheus/api/v1/query_range":
		queryRangeRequests.Inc()
		slowQueries = pathSlowQueryRange
```

Reply:
@AVESH-RAI Sorry, the above code is wrong. 
`slowQueries` is not thread-safe as a global variable here, it could be changed to a different counter by other query requests when increasing it's value.

Reply:
@Haleygo Thanks for the response.
From what I understand, if the `slowQueries` is a local scoped variable, it will be thread safe. 
Or is there any other way to make it thread-safe. Please let me know.

Reply:
Imaging the first request is to `prometheus/api/v1/query_range`, setting `slowQueries` to `pathSlowQueryRange`. During this request, a second request to `prometheus/api/v1/query` occurs, changing the global variable `slowQueries` to `pathSlowQuery`.
If the first request takes a long time and needs to be marked as slow, the counter is incorrectly added to `pathSlowQuery`.

>Or is there any other way to make it thread-safe. 

I don't see a good way except defining multiple counters to work separately.